src/app/connections/list/list.component.ts

```diff
@@ -1,20 +1,34 @@
-import { Component, Input } from '@angular/core';
+import { Component, /*Input,*/ OnInit } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
 
 import { Connections } from '../../store/connection/connection.model';
+import { ConnectionStore } from '../../store/connection/connection.store';
 
 @Component({
   selector: 'ipaas-connections-list',
   templateUrl: './list.component.html',
   styleUrls: ['./list.component.scss'],
 })
-export class ConnectionsListComponent {
+export class ConnectionsListComponent implements OnInit {
 
   truncateLimit = 80;
-
   truncateTrail = '…';
 
-  @Input() connections: Connections;
+  public connections: Observable<Connections>;
+  public loading: Observable<boolean>;
+
+  constructor(private store: ConnectionStore) {
+    //this.loading = store.loading;
+    this.connections = store.list;
+  }
+
+  updateConnections(event) {
+    console.log('Event from updateConnections: ' + JSON.stringify(event));
+    this.connections = event;
+  }
 
-  @Input() loading: boolean;
+  ngOnInit() {
+    this.store.loadAll();
+  }
 
```

Comment:
Because any directive it contains can inherit the data more easily. See [here](https://angular.io/docs/ts/latest/cookbook/component-communication.html).